akka-projection-core/src/main/scala/akka/projection/scaladsl/ProjectionManagement.scala

```diff
@@ -84,7 +91,22 @@ import akka.util.Timeout
     setOffset(projectionId, None)
 
   private def setOffset[Offset](projectionId: ProjectionId, offset: Option[Offset]): Future[Done] = {
-    topic(projectionId.name)
-      .ask(replyTo => Topic.Publish(SetOffset(projectionId, offset, replyTo)))
+    def askSetOffset(): Future[Done] = {
+      topic(projectionId.name)
+        .ask(replyTo => Topic.Publish(SetOffset(projectionId, offset, replyTo)))
+    }
+    retry(() => askSetOffset())
+  }
+
+  private def retry[T](operation: () => Future[T]): Future[T] = {
+    def attempt(remaining: Int): Future[T] = {
+      operation().recoverWith {
+        case e: TimeoutException =>
+          if (remaining > 0) attempt(remaining - 1)
+          else Future.failed(e)
+      }
+    }
+
+    attempt(retryAttempts)
```

Comment:
mainly because I only want to retry for TimeoutException
(since there is no delay or backoff it was also simple enough to implement)

There are probably no other exceptions than TimeoutException right now, but I'm considering changing to StatusReply.